extensions/ipynb/src/notebookSerializer.ts

```diff
@@ -94,3 +93,17 @@ export class NotebookSerializer implements vscode.NotebookSerializer {
 		return JSON.stringify(notebookContent, undefined, indentAmount);
 	}
 }
+
+export function detectIndent(jsonString: string) {
+	// ipynb is a JSON string of Object, hence first character will always `{`.
+	// Lets just take the distance between the first `{` and the next non-white space character`, ignoring \r & \n
+	if (!jsonString.startsWith('{')) {
+		return '';
+	}
+	// We're only interested in a small part of the string.
+	// The assumption is that we won't have an indentation of 10, just around 5 or so.
+	jsonString = jsonString.substring(1, 10).replace(/\r?\n/g, '');
+	// first index of non white space is the indentation.
+	const firstPositionOfNonWhiteSpace = jsonString.length - jsonString.trimStart().length;
+	return jsonString.substring(0, firstPositionOfNonWhiteSpace);
```

Comment:
If the file is `{ "a": 1 }` I think that would be parsed as one space of indentation by this code, but there is no indentation